crates/message_broker/src/channel.rs

```diff
@@ -29,17 +33,17 @@ impl RedisChannelSubscriber {
     }
 }
 
-impl ChannelSubscriber for RedisChannelSubscriber {
+impl ChannelSubscriber<Result<(), ()>> for RedisChannelSubscriber {
```

Comment:
`Result<(),()>` seems too hacky, what do you say if we replace it with an Enum or a boolean